wrappers/services.go

```diff
@@ -615,20 +648,27 @@ func (es *ensureSnapServicesContext) ensureSnapSystemdUnits(snapInfo *snap.Info,
 	}
 
 	// note that the Preseeding option is not used here at all
+	allGrps := quotaGroups.AllQuotaGroups()
 	for _, app := range snapInfo.Apps {
 		if !app.IsService() {
 			continue
 		}
```

Comment:
we could also split this into two loops, first finding all the services, and have a function that does the resolving for all the apps in one go and returns a mapping app-name -> service quota group or nil which means use opts.Quota group then have a looping doing the actual generation